storage/src/main/java/org/apache/kafka/server/log/internals/LogConfig.java

```diff
@@ -0,0 +1,504 @@
+    public static final long DEFAULT_RETENTION_BYTES = -1L;
+    public static final int DEFAULT_SEGMENT_INDEX_BYTES = 10 * 1024 * 1024;
+    public static final int DEFAULT_INDEX_INTERVAL_BYTES = 4096;
+    public static final long DEFAULT_FILE_DELETE_DELAY_MS = 60000L;
+    public static final String DEFAULT_CLEANUP_POLICY = TopicConfig.CLEANUP_POLICY_DELETE;
+    public static final long DEFAULT_FLUSH_MESSAGES_INTERVAL = Long.MAX_VALUE;
+    public static final long DEFAULT_FLUSH_MS = Long.MAX_VALUE;
+    public static final long DEFAULT_DELETE_RETENTION_MS = 24 * 60 * 60 * 1000L;
+    public static final long DEFAULT_MIN_COMPACTION_LAG_MS = 0;
+    public static final long DEFAULT_MAX_COMPACTION_LAG_MS = Long.MAX_VALUE;
+    public static final double DEFAULT_MIN_CLEANABLE_DIRTY_RATIO = 0.5;
+    public static final boolean DEFAULT_UNCLEAN_LEADER_ELECTION_ENABLE = false;
+    public static final int DEFAULT_MIN_IN_SYNC_REPLICAS = 1;
+    public static final String DEFAULT_COMPRESSION_TYPE = BrokerCompressionType.PRODUCER.name;
+    public static final boolean DEFAULT_PREALLOCATE = false;
+    public static final String DEFAULT_MESSAGE_TIMESTAMP_TYPE = "CreateTime";
+    public static final long DEFAULT_MESSAGE_TIMESTAMP_DIFFERENCE_MAX_MS = Long.MAX_VALUE;
+    public static final boolean DEFAULT_MESSAGE_DOWNCONVERSION_ENABLE = true;
+
+    public static final boolean DEFAULT_REMOTE_STORAGE_ENABLE = false;
+    public static final int DEFAULT_LOCAL_RETENTION_BYTES = -2; // It indicates the value to be derived from RetentionBytes
+    public static final int DEFAULT_LOCAL_RETENTION_MS = -2; // It indicates the value to be derived from RetentionMs
+    public static final List<String> DEFAULT_LEADER_REPLICATION_THROTTLED_REPLICAS = Collections.emptyList();
+    public static final List<String> DEFAULT_FOLLOWER_REPLICATION_THROTTLED_REPLICAS = Collections.emptyList();
+
+    /* See `TopicConfig.MESSAGE_FORMAT_VERSION_CONFIG` for details */
+    @Deprecated
+    public static final String DEFAULT_MESSAGE_FORMAT_VERSION = IBP_3_0_IV1.version();
+
+    // Leave these out of TopicConfig for now as they are replication quota configs
+    public static final String LEADER_REPLICATION_THROTTLED_REPLICAS_CONFIG = "leader.replication.throttled.replicas";
+    public static final String FOLLOWER_REPLICATION_THROTTLED_REPLICAS_CONFIG = "follower.replication.throttled.replicas";
+
+    @SuppressWarnings("deprecation")
+    private static final String MESSAGE_FORMAT_VERSION_CONFIG = TopicConfig.MESSAGE_FORMAT_VERSION_CONFIG;
+
+    // Visible for testing
+    public static final Set<String> CONFIGS_WITH_NO_SERVER_DEFAULTS = Collections.unmodifiableSet(Utils.mkSet(
+        TopicConfig.REMOTE_LOG_STORAGE_ENABLE_CONFIG,
+        TopicConfig.LOCAL_LOG_RETENTION_MS_CONFIG,
+        TopicConfig.LOCAL_LOG_RETENTION_BYTES_CONFIG,
+        LEADER_REPLICATION_THROTTLED_REPLICAS_CONFIG,
+        FOLLOWER_REPLICATION_THROTTLED_REPLICAS_CONFIG
```

Comment:
Added `LEADER_REPLICATION_THROTTLED_REPLICAS_CONFIG` and `FOLLOWER_REPLICATION_THROTTLED_REPLICAS_CONFIG` to this list since they never had server defaults.